bee-network/src/service/service.rs

```diff
@@ -0,0 +1,486 @@
+                internal_events,
+            } = receivers;
+
+            let (shutdown_tx1, shutdown_rx1) = oneshot::channel::<()>();
+            let (shutdown_tx2, shutdown_rx2) = oneshot::channel::<()>();
+            let (shutdown_tx3, shutdown_rx3) = oneshot::channel::<()>();
+
+            tokio::spawn(async move {
+                shutdown.await.expect("receiving shutdown signal");
+
+                shutdown_tx1.send(()).expect("receiving shutdown signal");
+                shutdown_tx2.send(()).expect("receiving shutdown signal");
+                shutdown_tx3.send(()).expect("receiving shutdown signal");
+            });
+            tokio::spawn(command_processor(
+                shutdown_rx1,
+                commands,
+                senders.clone(),
+                peerlist.clone(),
+            ));
+            tokio::spawn(event_processor(
+                shutdown_rx2,
+                internal_events,
+                senders.clone(),
+                peerlist.clone(),
+            ));
+            tokio::spawn(peer_checker(shutdown_rx3, senders, peerlist));
+
+            info!("Network service started.");
+        }
+    }
+}
+
+async fn command_processor(shutdown: Shutdown, commands: CommandReceiver, senders: Senders, peerlist: PeerList) {
+    debug!("Command handler running.");
+
+    let mut commands = ShutdownStream::new(shutdown, UnboundedReceiverStream::new(commands));
+
+    while let Some(command) = commands.next().await {
+        if let Err(e) = process_command(command, &senders, &peerlist).await {
+            error!("Error processing command. Cause: {}", e);
+            continue;
+        }
+    }
+
+    debug!("Command processor stopped.");
+}
+
+async fn event_processor(shutdown: Shutdown, events: InternalEventReceiver, senders: Senders, peerlist: PeerList) {
+    debug!("Event handler running.");
+
+    let mut int_events = ShutdownStream::new(shutdown, UnboundedReceiverStream::new(events));
+
+    while let Some(int_event) = int_events.next().await {
+        if let Err(e) = process_internal_event(int_event, &senders, &peerlist).await {
+            error!("Error processing internal event. Cause: {}", e);
+            continue;
+        }
+    }
+
+    debug!("Event processor stopped.");
+}
+
+// TODO: implement exponential back-off to not spam the peer with reconnect attempts.
+async fn peer_checker(shutdown: Shutdown, senders: Senders, peerlist: PeerList) {
+    debug!("Peer checker running.");
+
+    let Senders { internal_commands, .. } = senders;
+
+    // NOTE:
+    // We add a random amount of milliseconds to when the reconnector starts, so that even if 2 nodes
+    // go online at the same time the probablilty of them simultaneously dialing each other is reduced
+    // significantly.
+
+    let delay = Duration::from_millis(rand::thread_rng().gen_range(0u64..1000));
+    let start = Instant::now() + delay;
+    let period = Duration::from_secs(reconnect_interval_secs()); // `unwrap` is safe!
```

Comment:
An ghost `unwrap` :thinking:

Reply:
gone away through refactoring. Forgot to remove the comment.